src/base/utils/string.cpp

```diff
@@ -27,124 +27,74 @@
  * exception statement from your version.
  */
 
-#include <QByteArray>
-#include <QString>
-#include <QLocale>
-#include <cmath>
 #include "string.h"
 
-QString Utils::String::fromStdString(const std::string &str)
-{
-    return QString::fromUtf8(str.c_str());
-}
+#include <cmath>
 
-std::string Utils::String::toStdString(const QString &str)
-{
-    QByteArray utf8 = str.toUtf8();
-    return std::string(utf8.constData(), utf8.length());
-}
+#include <QByteArray>
+#include <QtGlobal>
+#include <QLocale>
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 2, 0))
+#include <QCollator>
+#endif
+#ifdef Q_OS_MAC
+#include <QThreadStorage>
+#endif
 
-// uses lessThan comparison
-bool Utils::String::naturalSort(const QString &left, const QString &right, bool &result)
+class NaturalCompare
```

Comment:
Put this class into anonymous namespace. You may also put the method implementations inside the class.